improver/fire_weather/canadian_forest_fire_weather_index.py

```diff
@@ -0,0 +1,98 @@
+# (C) Crown Copyright, Met Office. All rights reserved.
+#
+# This file is part of 'IMPROVER' and is released under the BSD 3-Clause license.
+# See LICENSE in the root of the repository for full licensing details.
+"""Plugin to calculate the Canadian Forest Fire Weather Index (FWI)."""
+
+import numpy as np
+from iris.cube import Cube
+
+from improver.fire_weather import FireWeatherIndexBase
+
+
+class CanadianForestFireWeatherIndex(FireWeatherIndexBase):
+    """
+    Plugin to calculate the Canadian Forest Fire Weather Index (FWI).
+
+    The FWI combines the Initial Spread Index (ISI) and the Build Up Index (BUI)
+    to provide a numerical rating of fire intensity. It represents the rate of
+    fire spread and the amount of available fuel.
+
+    This process is adapted directly from:
+        Equations and FORTRAN Program for the
+        Canadian Forest Fire Weather Index System
+        (C.E. Van Wagner and T.L. Pickett, 1985).
+        Pages 7-8, Equations 28-30.
+
+    Expected input units:
+        - Initial Spread Index (ISI): dimensionless
+        - Build Up Index (BUI): dimensionless
+    """
+
+    INPUT_CUBE_NAMES = ["initial_spread_index", "build_up_index"]
+    OUTPUT_CUBE_NAME = "canadian_forest_fire_weather_index"
+
+    initial_spread_index: Cube
+    build_up_index: Cube
+
+    def _calculate(self) -> np.ndarray:
+        """Calculate the Fire Weather Index (FWI).
+
+        From Van Wagner and Pickett (1985), Page 8: Steps 4-6
+
+        Returns:
+            np.ndarray: The calculated FWI values.
```

Comment:
```suggestion
            The calculated FWI values.
```
Return type in docstring not required. Sphinx/Read the Docs infers from type hint.